### PR TITLE
Configure Renovatebot

### PR DESCRIPTION
**Pull request by quitterie-lcs**
## Purpose

We decided to migrate to `renovate` to manage all our dependencies on `openfun` projects. Till now, `pyup` was the bot used to automatically upgrade dependencies.

## Proposal

- [x] Remove `pyup` configuration file
- [x] Add `openfun` `renovate` preset configuration

### CODE REVIEW

**Review by jbpenrath — COMMENTED** [review]
You should search and remove `# pyup: ignore`.

Then if you want renovate also ignores dependencies previously ignored by pyup, you can add into the `renovate.json` file:

```json
"packageRules": [
    {
      "enabled": false,
      "groupName": "ignored python dependencies",
      "matchManagers": [
        "setup-cfg"
      ],
      "matchPackageNames": [
        "deps1",
        "deps2",
        "..."
      ]
    }
  ]
```

File: Dockerfile [hunk 1]
```diff
@@ -32,13 +32,13 @@ WORKDIR /app
...
-      libpango-1.0-0 \
-      libpangocairo-1.0-0 \
-      shared-mime-info && \
+    gettext \
+    libcairo2 \
+    libffi-dev \
+    libgdk-pixbuf2.0-0 \
+    libpango-1.0-0 \
+    libpangocairo-1.0-0 \
+    shared-mime-info && \
```
Comment (jmaupetit) [inline Dockerfile:41]: I find the original indentation more relevant.

File: Dockerfile [hunk 2]
```diff
@@ -75,12 +75,12 @@ ENV PYTHONUNBUFFERED=1
 RUN pip uninstall -y marion
 COPY ./src/marion /usr/local/src/marion
 RUN cd /usr/local/src/marion && \
-      pip install -e .[dev,sandbox]
+    pip install -e .[dev,sandbox]
```
Comment (jmaupetit) [inline Dockerfile:78]: ditto

File: Dockerfile [hunk 3]
```diff
@@ -75,12 +75,12 @@ ENV PYTHONUNBUFFERED=1
...
 
 # Copy extra packages
 COPY ./src/howard /usr/local/src/howard
 RUN cd /usr/local/src/howard && \
-      pip install -e .
+    pip install -e .
```
Comment (jmaupetit) [inline Dockerfile:83]: ditto

File: Dockerfile [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # Marion, the documents factory
 
 # -- Base image --
-FROM python:3.8-slim as base
+FROM python:3.8.11-slim as base
```
Comment (jmaupetit) [inline Dockerfile:4]: I don't think it's required now.

File: .circleci/config.yml [hunk 5]
```diff
@@ -18,7 +18,7 @@ jobs:
   # Check that the git history is clean and complies with our expectations
   lint-git:
     docker:
-      - image: python:3.8-buster
+      - image: python:3.8-bullseye
```
Comment (jmaupetit) [inline .circleci/config.yml:21]: For consistency with the project's image, have you tried to use the `3.8-slim` image in the CI?

Comment (quitterie-lcs) [reply .circleci/config.yml:21]: It doesn't exist

File: Dockerfile [hunk 6]
```diff
@@ -39,7 +39,7 @@ RUN apt-get update && \
       libpango-1.0-0 \
       libpangocairo-1.0-0 \
       shared-mime-info && \
-    rm -rf /var/lib/apt/lists/*
+      rm -rf /var/lib/apt/lists/*
```
Comment (jbpenrath) [inline Dockerfile:42]: Linter on save? I think previous indentation was better.

File: .circleci/config.yml (hunk 5 above)
Comment (quitterie-lcs) [reply .circleci/config.yml:21]: That's relevant, indeed !

**Review by jbpenrath — APPROVED** [review]
LGTM ✅ !

File: Dockerfile (hunk 6 above)
Comment (quitterie-lcs) [reply Dockerfile:42]: Always the same guilty one :laughing:

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**